### PR TITLE
feat(microplanning): add work area status metrics to map header

### PR DESCRIPTION
**Pull request by Charl1996**
## Product Description

Adds 6 new metric cards to the microplanning map header alongside the existing "Days Remaining" card:

- **Unvisited Work Areas** — count + % of non-excluded WAs with zero approved `UserVisit` records
- **Visited Work Areas** — count + % of non-excluded WAs with at least one approved `UserVisit`
- **EVC Reached** — count + % of non-excluded WAs whose approved-visit count is ≥ their `expected_visit_count`
- **Inaccessible Work Areas** — count + % of non-excluded WAs with status `INACCESSIBLE`
- **Excluded Work Areas** — count + % of all WAs with status `EXCLUDED`
- **% WA visited to % total visits** — ratio of (% non-excluded WAs visited) to (% of total expected visits completed, measured via approved visits), displayed as a percentage itself (as per requirement). Think of this metric as "how spread the completed work is" across all work areas.

Each count card displays the raw count on the left and percentage on the right of the card (e.g. `20    44%`).

image

## Technical Summary
RP 3
Jira: [CCCT-2323](https://dimagi.atlassian.net/browse/CCCT-2323)

- Extended `get_metrics_for_microplanning()` in `commcare_connect/microplanning/views.py` to drive the visited/unvisited/EVC metrics from actual approved `UserVisit` records. A `Subquery` annotates each `WorkArea` with an `approved_count` (count of `UserVisit` rows for that work area with `status=approved`), and a single `.aggregate()` computes all buckets.
- `Inaccessible` and `Excluded` still read from `WorkArea.status`, since those are operational states not implied by visit activity.
- The `% WA visited to % total visits` ratio basically tells you how "spread" the visits are across all work areas (it's required at this stage to be a percentage - real values are expected to hover around 100). Orphan approved visits (visits with `work_area IS NULL`) are excluded from the denominator so they don't deflate the ratio.
- All metrics remain filter-independent — map date filters do not affect the header numbers.
- Template uses a `unit: "%"` key on the ratio metric to avoid a fragile translated-string comparison.

### Safety story

Read-only change — no writes, no model changes, no migrations. The new metrics are computed from existing `WorkArea` and `UserVisit` rows via ORM aggregation. Zero/empty-data edge cases (no work areas, all excluded, zero expected visits, zero approved visits) all return `"--"` or `0` safely.

### Automated test coverage

Added `TestGetMetricsForMicroplanningWorkAreas` in `commcare_connect/microplanning/tests/test_views.py` covering:
- Each status bucket (unvisited, visited, EVC reached, inaccessible, excluded) driven by approved `UserVisit` records
- Correct percentage denominators (non-excluded vs total)
- `% WA visited to % total visits` ratio calculation from approved visits
- Orphan approved visits (no `work_area`) are excluded from the ratio denominator
- Zero-denominator edge cases (`"--"` fallback)
- All-excluded work areas edge case

### QA Plan
N.A

### Labels & Review

- [ ] The set of people pinged as reviewers is appropriate for the level of risk of the change

[CCCT-2323]: https://dimagi.atlassian.net/browse/CCCT-2323?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: commcare_connect/microplanning/views.py [hunk 1]
```diff
@@ -130,11 +130,71 @@ def microplanning_home(request, *args, **kwargs):
...
+
+    def pct(numerator, denominator):
+        if not denominator:
+            return None
+        return round(numerator / denominator * 100)
+
+    total_expected = agg["total_expected_visits"] or 0
+    if non_excluded and total_expected:
+        total_visits = UserVisit.objects.filter(opportunity=opportunity).count()
+        pct_wa_visited = agg["visited"] / non_excluded
```
Comment (pxwxnvermx) [inline commcare_connect/microplanning/views.py]: nit: Can you please add the same `agg or 0` check for this `agg["visited"]` it would make it similar to the other values.

**Review by pxwxnvermx — DISMISSED** [review]
just a small nit. rest of the changes look good. Thanks

File: commcare_connect/microplanning/views.py (hunk 1 above)
Comment (Charl1996) [reply commcare_connect/microplanning/views.py]: See latest commits.

Comment (Charl1996) [reply commcare_connect/microplanning/views.py]: https://github.com/dimagi/commcare-connect/pull/1143/commits/d4ead2c047b669d8f652f45c0e61fd3af7cb0d95

Comment (Charl1996) [reply commcare_connect/microplanning/views.py]: Fix a bug in https://github.com/dimagi/commcare-connect/pull/1143/commits/7d587c6f2ba5e62260c978bf95ae0f8c6b5a8fda regarding evaluation order. (also, this commit converts the last metric to percentage as per the requirement)

File: commcare_connect/microplanning/tests/test_views.py [hunk 2]
```diff
@@ -932,3 +912,245 @@ def test_work_started_statuses_are_skipped(self, client, org_user_admin, opportu
...
+
+    def test_end_date_missing(self, opp):
+        """No end_date → Days Remaining shows '--'."""
+
+        opp.end_date = None
+        metrics = get_metrics_for_microplanning(opp)
+        m = self._get_metric(metrics, "Days Remaining")
+        assert m["value"] == "--"
+
+    def test_end_date_in_past(self, opp):
```
Comment (zandre-eng) [inline commcare_connect/microplanning/tests/test_views.py]: Nit: These two tests can be parameterized together.

**Review by zandre-eng — DISMISSED** [review]

File: commcare_connect/microplanning/tests/test_models.py
Comment (zandre-eng) [inline commcare_connect/microplanning/tests/test_models.py:11]: Nit: All the tests in this class look similar enough that they could be parametrized.

File: commcare_connect/microplanning/views.py [hunk 3]
```diff
@@ -151,11 +151,90 @@ def microplanning_home(request, *args, **kwargs):
...
+            return None
+        return round(numerator / denominator * 100)
+
+    total_expected = agg["total_expected_visits"] or 0
+    if non_excluded_count and total_expected:
+        total_approved_visits = UserVisit.objects.filter(
+            opportunity=opportunity,
+            status=VisitValidationStatus.approved,
+            work_area__isnull=False,
+        ).count()
```
Comment (zandre-eng) [inline commcare_connect/microplanning/views.py]: Nit: Instead of doing a separate DB query here you could do this aggregation in the `agg` query with `total_approved_visits=Sum("approved_count", filter=non_excluded)` and then here you can simply retrieve it with `total_approved_visits = agg["total_approved_visits"] or 0`

**Review by zandre-eng — APPROVED** [review]
Minor, non-blocking nits, but otherwise looks good.

File: commcare_connect/microplanning/views.py (hunk 3 above)
Comment (Charl1996) [reply commcare_connect/microplanning/views.py]: https://github.com/dimagi/commcare-connect/pull/1143/commits/137bb11f54c56428022a1cc8cf7dd5a3346f4915

File: commcare_connect/microplanning/tests/test_models.py
Comment (Charl1996) [reply commcare_connect/microplanning/tests/test_models.py:11]: https://github.com/dimagi/commcare-connect/pull/1143/commits/137bb11f54c56428022a1cc8cf7dd5a3346f4915

**Review by zandre-eng — APPROVED** [review]
